src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -2675,6 +2675,135 @@ static void Main(string[] args)
                 Diagnostic(ErrorCode.ERR_CantInferMethTypeArgs, "F").WithArguments("Program.F<T>(System.Func<T[]>)").WithLocation(9, 17));
         }
 
+        [Fact]
+        public void TypeInference_Spread_01()
+        {
+            string source = """
+                class Program
+                {
+                    static T[] F<T>(T[] arg) => arg;
+                    static void Main()
+                    {
+                        int[] x = [1, 2];
+                        int[] y = F([..x]);
+                        y.Report();
+                    }
+                }
+                """;
+            CompileAndVerify(new[] { source, s_collectionExtensions }, expectedOutput: "[1, 2], ");
+        }
+
+        [Fact]
+        public void TypeInference_Spread_02()
+        {
+            string source = """
+                class Program
+                {
+                    static T[] F<T>(T[] arg) => arg;
+                    static void Main()
+                    {
+                        object[] x = [1, 2];
+                        var y = F([..x, 3]);
+                        y.Report(includeType: true);
+                    }
+                }
+                """;
+            CompileAndVerify(new[] { source, s_collectionExtensions }, expectedOutput: "(System.Object[]) [1, 2, 3], ");
+        }
+
+        [Fact]
+        public void TypeInference_Spread_03()
+        {
+            string source = """
+                class Program
+                {
+                    static T[] F<T>(T[] arg) => arg;
+                    static void Main()
+                    {
+                        int[] x = [1, 2];
+                        var y = F([..x, (object)3]);
+                        y.Report(includeType: true);
+                    }
+                }
+                """;
+            CompileAndVerify(new[] { source, s_collectionExtensions }, expectedOutput: "(System.Object[]) [1, 2, 3], ");
+        }
+
+        // If we allow inference from a spread element _expression_ rather than simply
```

Comment:
would def want to do this eventually. I feel like if we can make `return [null]` "just work" then `return [..[null]]` should also just work.